_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: post
+name: docker-meets-continuous-deployment
+title: Docker meets Continuous Deployment
+date: 2016-03-01 00:00:00 +00:00
+author: Robert Firek
+image:
+    src: /assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment.png
+    attribution:
+        text: Track icon by Freepik (CC 3.0 BY)
+        href: http://www.freepik.com
+tags:
+- Continuous Deployment
+- Docker
+- TeamCity
+--- 
+
+About one year ago I had a first contact with Docker. This new kind on the block promised to relieve our poor computers 
+of installation of all tools, languages, dependencies and operating systems. Isolated run environments emerged on developers computers.
+
+A conservative approach of my ops teammates didn't change my joy of using Docker. Despite many people described Docker
+as tool writen by developers for development our industry found new ways of using images and containers. Images of our 
+applications and services became units of deployment for tool like  Kubernetes, Docker Swarm or Marathon.
+
+But... How this images appear in these tools?
+
+## Set The Scene
+
+From developers perspective any application is manifested by a code. There is long way before we finally find some place on a production environment.
+To show this process we need small application. Our application is going to expose some HTTP API which we are going to use after the deployment. 
+Let's assume that will are going to use **Gradle** to build the application and **TeamCity** as build automation tool.
+
+On each **TeamCity** build agent we should have installed **Docker**. We will also use this machine to deploy our application.
+In the real project probably you don't want to do that, but the deployment phase can be changed to fit better to your environment.
```

Comment:
OK after reading until the end, I understand that it's not to "deploy" the application that you will use the machine, but it is actually to "run" the application. It would probably be more clear to use "run" instead of "deploy" here.